src/main/resources/com/owlcyberdefense/mil-std-2045/xsd/milstd2045.enums.dfdl.xsd

```diff
@@ -0,0 +1,1067 @@
+        xmlns="http://www.w3.org/2001/XMLSchema"
+        xmlns:fn="http://www.w3.org/2005/xpath-functions"
+        elementFormDefault="unqualified">
+
+  <import namespace="urn:milstd2045DFDLInternal"
+          schemaLocation="com/owlcyberdefense/mil-std-2045/xsd/milstd2045.internal.dfdl.xsd"/>
+
+  <annotation>
+    <appinfo source="http://www.ogf.org/dfdl/">
+      <dfdl:format lengthKind='implicit' ref="msi:msBase"/>
+    </appinfo>
+  </annotation>
+
+  <simpleType name="checkedInt" dfdl:lengthKind="explicit">
+    <annotation>
+      <appinfo source="http://www.ogf.org/dfdl/">
+        <dfdl:assert>{ dfdl:checkConstraints(.) }</dfdl:assert>
+      </appinfo>
+    </annotation>
+    <restriction base="xs:unsignedInt"/>
+  </simpleType>
+
+  <simpleType name="enumBase">
+    <restriction base="tns:checkedInt"/>
+  </simpleType>
+
+  <simpleType name="enum1" dfdl:length="1">
+    <xs:restriction base="tns:enumBase"/>
+  </simpleType>
+
+  <simpleType name="enum2" dfdl:length="2">
+    <xs:restriction base="tns:enumBase"/>
+  </simpleType>
+
+  <simpleType name="enum3" dfdl:length="3">
+    <xs:restriction base="tns:enumBase"/>
+  </simpleType>
+
+  <simpleType name="enum4" dfdl:length="4">
+    <xs:restriction base="tns:enumBase"/>
+  </simpleType>
+
+  <simpleType name="enum5" dfdl:length="5">
+    <xs:restriction base="tns:enumBase"/>
+  </simpleType>
+
+  <simpleType name="enum6" dfdl:length="6">
+    <xs:restriction base="tns:enumBase"/>
+  </simpleType>
+
+  <simpleType name="enum9" dfdl:length="9">
+    <xs:restriction base="tns:enumBase"/>
+  </simpleType>
+
+  <!--
+    Workaround for DAFFODIL-2596
+    for the enum restiction base, must specify delimited
+   -->
+  <xs:simpleType name="enumString" dfdl:lengthKind="delimited">
+    <xs:restriction base="xs:string"/>
+  </xs:simpleType>
+
+  <!--
+  Used to create invalid element values on purpose
+  -->
+  <simpleType name="invalidEnum">
+    <restriction base="xs:string">
+      <pattern value="OK_.*"/>
+    </restriction>
+  </simpleType>
+
+  <simpleType name="versionEnum" dfdlx:repType="tns:enum4">
+    <restriction base="tns:enumString">
+      <enumeration value="C" dfdlx:repValues="2"/>
+      <enumeration value="D1" dfdlx:repValues="4"/>
+    </restriction>
+  </simpleType>
+
+  <simpleType name="dataCompressionEnum" dfdlx:repType="tns:enum2">
+    <restriction base="tns:enumString">
+      <enumeration value="unix_compress_uncompress" dfdlx:repValues="0"/>
+      <enumeration value="GZIP" dfdlx:repValues="1"/>
+      <pattern value="OK_*"/>
+    </restriction>
+  </simpleType>
+
+  <simpleType name="umfEnum" dfdlx:repType="tns:enum4">
+    <restriction base="tns:enumString">
+      <enumeration value="OK_Link_16" dfdlx:repValues="0"/>
+      <enumeration value="OK_Binary_File" dfdlx:repValues="1"/>
+      <enumeration value="OK_VMF" dfdlx:repValues="2"/>
+      <enumeration value="OK_NITFS" dfdlx:repValues="3"/>
+      <enumeration value="OK_RDM" dfdlx:repValues="4"/>
```

Comment:
OK_RDM doesn't exist in ms2hC, looks like it's only ms2hDC1. Comparing Table IV of the spec, it looks like the codes differ for ms2hC and ms2hDC1. So I think we need 2 different enums. This applies for the other enums as well, as they also appear to have different definitions between the 2 versions

Reply:
Just a name change from C to D1. Doesn't change validity of data.